terraform/templates/backend.tf.gotmpl

```diff
@@ -1,3 +1,46 @@
+{{if (.Env.ENABLE_LOCALSTACK | strings.Contains "1")}}
+terraform {
+  backend "local" {}
+}
+
+provider "aws" {
+  profile                     = var.aws_profile
+  region                      = var.aws_region
+  s3_force_path_style         = true
+  secret_key                  = "mock_secret_key"
+  skip_credentials_validation = true
+  skip_metadata_api_check     = true
+  skip_requesting_account_id  = true
+
+  endpoints {
+    apigateway     = "{{ .Env.LOCALSTACK_ENDPOINT }}"
+    cloudformation = "{{ .Env.LOCALSTACK_ENDPOINT }}"
+    cloudwatch     = "{{ .Env.LOCALSTACK_ENDPOINT }}"
+    ec2            = "{{ .Env.LOCALSTACK_ENDPOINT }}"
+    dynamodb       = "{{ .Env.LOCALSTACK_ENDPOINT }}"
+    es             = "{{ .Env.LOCALSTACK_ENDPOINT }}"
+    firehose       = "{{ .Env.LOCALSTACK_ENDPOINT }}"
+    iam            = "{{ .Env.LOCALSTACK_ENDPOINT }}"
+    kinesis        = "{{ .Env.LOCALSTACK_ENDPOINT }}"
+    lambda         = "{{ .Env.LOCALSTACK_ENDPOINT }}"
+    route53        = "{{ .Env.LOCALSTACK_ENDPOINT }}"
+    redshift       = "{{ .Env.LOCALSTACK_ENDPOINT }}"
+    s3             = "{{ .Env.LOCALSTACK_ENDPOINT }}"
+    secretsmanager = "{{ .Env.LOCALSTACK_ENDPOINT }}"
+    ses            = "{{ .Env.LOCALSTACK_ENDPOINT }}"
+    sns            = "{{ .Env.LOCALSTACK_ENDPOINT }}"
+    sqs            = "{{ .Env.LOCALSTACK_ENDPOINT }}"
+    ssm            = "{{ .Env.LOCALSTACK_ENDPOINT }}"
+    stepfunctions  = "{{ .Env.LOCALSTACK_ENDPOINT }}"
+    sts            = "{{ .Env.LOCALSTACK_ENDPOINT }}"
+  }
+}
+{{else}}
+provider "aws" {
```

Comment:
This section should probably be also under some conditional, so it is still backwards-compatible to the environments that have that config in checked-in main.tf. maybe something like "ICMK_LEGACY" or "ICMK_FUTURE"

@igorkotof Thougths?

Reply:
Let's implement it in the next stories